src/cplscheme/CouplingScheme.hpp

```diff
@@ -173,6 +163,17 @@ class CouplingScheme {
   /// @brief Returns true, if data has been exchanged in last call of advance().
   virtual bool hasDataBeenReceived() const = 0;
 
+  // @todo find a better name. This is too low level.
+  virtual void storeTimeStepReceiveData(double relativeDt) = 0;
+
+  // @todo find a better name. This is too low level.
+  /**
+   * @brief retreives time step data from CouplingData into mesh values
+   *
+   * @param relativeDt relative dt associated with the data.
+   */
+  virtual void retreiveTimeStepReceiveData(double relativeDt) = 0;
```

Comment:
Can we also use `exchangeSecondData()` here?

Reply:
Situation improved a lot, but still looks like we need a special treatment for `SerialCouplingScheme`. 0aa0a6899fc0610496c5e720388118cdf92eea50